packages/hash/frontend/src/blocks/page/createCommentPlugin.tsx

```diff
@@ -0,0 +1,141 @@
+import { TextToken } from "@hashintel/hash-shared/graphql/types";
+import { Popper } from "@mui/material";
+import { Schema } from "prosemirror-model";
+import { Plugin, PluginKey } from "prosemirror-state";
+import { ensureMounted } from "../../lib/dom";
+import { CommentTextField } from "./Comments/CommentTextField";
+import { RenderPortal } from "./usePortals";
+
+export type CreateCommentAction =
+  | {
+      type: "open";
+      payload: {
+        anchorNode: HTMLElement;
+        blockId: string;
+        onSubmit: (content: TextToken[]) => Promise<void>;
+      };
+    }
+  | { type: "close" };
+
+interface CreateCommentState {
+  open: boolean;
+  anchorNode: HTMLElement | null;
+  blockId: string | null;
+  onSubmit: ((content: TextToken[]) => Promise<void>) | null;
+}
+
+export const createCommentPluginKey = new PluginKey<CreateCommentState, Schema>(
+  "createComment",
+);
+
+const initState = {
+  open: false,
+  anchorNode: null,
+  blockId: null,
+  onSubmit: null,
+};
+
+export const createCommentPlugin = (
```

Comment:
I'm using a plugin here so that the `CommentInput` is rendered through `renderPortal` at the root of the page. This makes it so the `CommentInput`'s Prosemirror document is not inside the page's document, preventing a lot of interactions that would have to be worked around.  I think keeping them isolated will keep things simpler.

Reply:
Yeah that's valid. I went with a plugin because the bahaviour I wanted was very close to `createSuggester`. But I moved it out of a plugin in 718f3a4faf390c9b82f62579be285b1297f2340b and it really simplified things, thanks!

Reply:
On second thought, even though it simplified the component a lot, I forgot that in the future we will add inline text comments, and having it be a plugin would be the way to go in that case (because we would have to position it where the text is and the event to open the comment input will come from the format text plugin). Having the comment input be a component for block comments and a plugin for inline comments seems silly to me, so I think we might want to change this back. What do you think?

Reply:
I think we can change it to a plugin later if that becomes necessary.